GPR183.py

```diff
@@ -0,0 +1,184 @@
+# Importing modules
+import os
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from typing import Union
+
+
+def fastq_filter(input_path: str = None, output_filename: str = None, *,
+                 gc_bound: Union[tuple, int, float] = (0, 100),
+                 length_bound: Union[tuple, int, float] = (0, 2**32),
+                 quality_threshold: Union[int, float] = 0) -> None:
+    """
+    This function work with FASTQ files and filters them by
+    GC content, length and Q-score.
+
+    Arguments (positional):
+    - input_path (str): full path to the file that you want to work with
+    - output_filename (str): enter just a name of the file, don't add extention
+
+    Arguments (keyword):
+    - gc_bound (tuple, int, float): tuple of required range of GC percentage (inclusive),
+    num or float if only higher border of the range is needed (exclusive).
+    - length_bound (tuple, int, float): tuple of required range of sequences length (inclusive),
+    num or float if only higher border of the range is needed (exclusive).
+    - quality_threshold (int): int of lowest level of Q-score (inclusive).
+
+    Output:
+    - list of BioSeq records. Write file to .fastq
+    """
+    # Chech if PATH for input file is given
+    if input_path is None:
+        raise ValueError("You didn't enter any PATH to file")
+    # Chech if folder exist and create outout_path if not given
+    input_folder = input_path.rsplit('/', 1)[0]
+    input_name = input_path.rsplit('/', 1)[1]
+    is_exist = os.path.exists(f'{input_folder}/fastq_filtrator_resuls/')
+    if not is_exist:
+        os.makedirs(f'{input_folder}/fastq_filtrator_resuls/')
+    if output_filename is None:
+        output_path = f'{input_folder}/fastq_filtrator_resuls/{input_name}'
+    else:
+        output_path = f'{input_folder}/fastq_filtrator_resuls/{output_filename}.fastq'
+    # Create dict from FASTQ
+    seqs = list(SeqIO.parse(input_path, "fastq"))
```

Comment:
Всё указано супер корректно, но не совсем понятно, зачем нам полученный результат переводить в лист? 
При использовании парсера из BioPython, мы уже получаем коллекцию из SeqRecord, по которой можем итерироваться. У меня вот так работает :)
```suggestion
    seqs = SeqIO.parse(input_path, "fastq")
```

Reply:
Ровно так